tests/rest/auth/test_oidc.py

```diff
@@ -437,46 +437,106 @@ def test_oidc_device_flow_without_pkce_nor_secret(requests_mock, caplog):
+            True,
+        ),
+        (
             "use PKCE",
             "myclient", True, None,
-            {"scope": "df openid", "code_challenge": True, "code_verifier": True}
-    ),
-    (
+            {"scope": "df openid", "code_challenge": True, "code_verifier": True},
+            False,
+        ),
+        (
+            "use PKCE",
+            "myclient", True, None,
+            {"scope": "df openid", "code_challenge": True, "code_verifier": True},
+            True,
+        ),
+        (
             "auto PKCE",
             "myclient", None, None,
-            {"scope": "df openid", "code_challenge": ABSENT, "code_verifier": ABSENT}
-    ),
-    (
+            {"scope": "df openid", "code_challenge": ABSENT, "code_verifier": ABSENT},
+            False,
+        ),
+        (
+            "auto PKCE",
+            "myclient", None, None,
+            {"scope": "df openid", "code_challenge": ABSENT, "code_verifier": ABSENT},
+            True,
+        ),
+        (
+            "auto PKCE, default client with PKCE",
+            "default-with-pkce", None, None,
+            {"scope": "df openid", "code_challenge": True, "code_verifier": True},
+            False,
+        ),
+        (
             "auto PKCE, default client with PKCE",
             "default-with-pkce", None, None,
-            {"scope": "df openid", "code_challenge": True, "code_verifier": True}
-    ),
-    (
+            {"scope": "df openid", "code_challenge": True, "code_verifier": True},
+            True,
+        ),
+        (
+            "auto PKCE, default client no PKCE",
+            "default-no-pkce", None, None,
+            {"scope": "df openid", "code_challenge": ABSENT, "code_verifier": ABSENT},
+            False,
+        ),
+        (
             "auto PKCE, default client no PKCE",
             "default-no-pkce", None, None,
-            {"scope": "df openid", "code_challenge": ABSENT, "code_verifier": ABSENT}
-    ),
-    (
+            {"scope": "df openid", "code_challenge": ABSENT, "code_verifier": ABSENT},
+            True,
+        ),
+        (
             "auto PKCE, default client with PKCE, and secret",
             "default-with-pkce", None, "$3cr3t",
-            {"scope": "df openid", "client_secret": "$3cr3t", "code_challenge": ABSENT, "code_verifier": ABSENT}
-    ),
-])
+            {"scope": "df openid", "client_secret": "$3cr3t", "code_challenge": ABSENT, "code_verifier": ABSENT},
+            False,
+        ),
+        (
+            "auto PKCE, default client with PKCE, and secret",
+            "default-with-pkce", None, "$3cr3t",
+            {"scope": "df openid", "client_secret": "$3cr3t", "code_challenge": ABSENT, "code_verifier": ABSENT},
+            True,
+        ),
+    ],
+)
+# fmt: on
 def test_oidc_device_flow_auto_detect(
-        requests_mock, caplog, mode, client_id, use_pkce, client_secret, expected_fields
+    requests_mock,
```

Comment:
There are also a couple of other tests above that could be covered

Reply:
Indeed. 
I will continue with your branch that has these covered.